tests/benchmarks/exercism_practice.py

```diff
@@ -138,7 +139,7 @@ async def run_exercise(problem_dir, language="python", max_iterations=2):
     )
 
     iterations = 0
-    while iterations < max_iterations:
+    while iterations < max_iterations and client.started:
```

Comment:
If it exits because of the session turning off (presumably because of an error), should we mark the test reason for failure as 'error'? I think it makes sense, but was wondering what y'all might think

Reply:
We should definitely record failures due to rate limiting or api issues separately from other failures